proxlb

```diff
@@ -420,11 +487,38 @@ def get_vm_statistics(api_object, ignore_vms, balancing_type):
                     vm_statistics[vm['name']]['disk_used']      = vm['disk']
                     vm_statistics[vm['name']]['vmid']           = vm['vmid']
                     vm_statistics[vm['name']]['node_parent']    = node['node']
-                    vm_statistics[vm['name']]['type']           = 'vm'
-                    # Rebalancing node will be overwritten after calculations.
-                    # If the vm stays on the node, it will be removed at a
-                    # later time.
                     vm_statistics[vm['name']]['node_rebalance'] = node['node']
+                    vm_statistics[vm['name']]['storage']        = {}
+                    vm_statistics[vm['name']]['type']           = 'vm'
+
+                    # Get disk details of the related object.
+                    _vm_details = api_object.nodes(node['node']).qemu(vm['vmid']).config.get()
+                    logging.info(f'{info_prefix} Getting disk information for vm {vm["name"]}.')
+
+                    for vm_detail_key, vm_detail_value in _vm_details.items():
+                        # vm_detail_key_validator = re.sub('\d+$', '', vm_detail_key)
+                        vm_detail_key_validator = re.sub(r'\d+$', '', vm_detail_key)
+
+                        if vm_detail_key_validator in _vm_details_storage_allowed:
+                            vm_statistics[vm['name']]['storage'][vm_detail_key] = {}
+                            match = re.match(r'([^:]+):[^/]+/(.+),iothread=\d+,size=(\d+G)', _vm_details[vm_detail_key])
+
+                            # Create an efficient match group and split the strings to assign them to the storage information.
+                            if match:
+                                _volume    = match.group(1)
+                                _disk_name = match.group(2)
+                                _disk_size = match.group(3)
+
+                                vm_statistics[vm['name']]['storage'][vm_detail_key]['name']               = _disk_name
+                                vm_statistics[vm['name']]['storage'][vm_detail_key]['device_name']        = vm_detail_key
+                                vm_statistics[vm['name']]['storage'][vm_detail_key]['volume']             = _volume
+                                vm_statistics[vm['name']]['storage'][vm_detail_key]['storage_parent']     = _volume
+                                vm_statistics[vm['name']]['storage'][vm_detail_key]['storage_rebalance']  = _volume
+                                vm_statistics[vm['name']]['storage'][vm_detail_key]['size']               = _disk_size[:-1]
+                                logging.info(f'{info_prefix} Added disk for {vm["name"]}: Name {_disk_name} on volume {_volume} with size {_disk_size}.')
+                            else:
+                                logging.info(f'{info_prefix} No disks for {vm["name"]} found.')
```

Comment:
Validate if really no disk was found or the disk type is simply unsupported (e.g. CD-ROM, ...).

(just not to self, to fix in the evening)